sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/Auction.java

```diff
@@ -80,14 +80,14 @@ public Auction decode(
 
   /** Extra auction properties. */
   @JsonProperty
-  private final String itemName;
+  public final String itemName;
```

Comment:
<!--thread_id:cc_150930883_t; commit:5d81ac0253726b036b7f1264e3719ee7507fedf7; resolved:0-->
<!--section:context-quote-->
> **kennknowles** wrote:
> I mentioned `@AutoValue` before and I think you can do this, at least for Jackson:
> 
> ```java
> @AutoValue
> @AutoValue.CopyAnnotations
> public abstract class Auction {
>   @JsonProperty
>   public abstract String getItemName();
> }
> ```
> 
> I don't know that it is possible to use `@AutoValue` with avro schema generation, though.

<!--section:body-->
I don't see a straightforward way to do this:
 - Avro seems to require a no-args constructor in our case, i.e. we seem to end up here when creating records instances: https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/specific/SpecificData.java#L351 
 - AutoValue doesn't support generation of no-args constructor, from what I can see;

Probably we could work around this by having Avro-specific subclasses of the model classes, but I don't think it should be done as part of this effort.